File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> um;
        vector<int> vec;
        int num;
        
        // 숫자를 키로, 인덱스를 값으로 저장
        for (int i = 0; i < nums.size(); i++) {
            num = nums[i];
            um[num] = i;
        }
        
        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            
            // `complement`가 `nums`에 있고, 인덱스가 다를 때
            if (um.find(complement) != um.end() && um[complement] != i) {
                vec.push_back(i);
                vec.push_back(um[complement]);
                return vec;
            }
        }
        return vec; // 해결책이 없는 경우 빈 벡터 반환
    }
};

File: Data Structures & Algorithms/search-insert-position/submission-0.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int l=0, r=nums.length-1;

        while(l<=r){
            int index = (l+r)/2;

            if(nums[index] < target){
                l = index+1;
            }else if(nums[index]>target){
                r = index-1;
            }else{
                return index;
            }
        }
        return l;
    }
}